### PR TITLE
interfaces: revamp overview page

### PR DESCRIPTION
**Pull request by swhite2**
(eventually) replaces the old overview page, with the intent of quicker debugging by presenting the most relevant information that could indicate faulty configuration.

last remaining bit is cleaning up `get_interfaces_info()`, but the `interface_list` widget is still using a callout, this should be replaced by the already present and adjusted callback mechanism in `src/www/widgets/api/plugins/interfaces.inc`.

This commit omits some of the link-specific information such as ppp uptime, disconnect/release mechanism, as well as wireless and bridge information. Since there is more of this type of information available than was originally being handled by `get_interfaces_info()`, perhaps it makes more sense to extend the backend script with the relevant bits in time.

### CODE REVIEW

File: src/opnsense/scripts/interfaces/ifinfo.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
+#!/usr/local/bin/php
```
Comment (fichtner) [inline src/opnsense/scripts/interfaces/ifinfo.php:1]: Would be nicer to fold this into pluginctl, very similar functionality already exists there.

File: src/www/widgets/api/plugins/interfaces.inc [hunk 2]
```diff
@@ -30,37 +30,31 @@
...
-                $interfaceItem['status'] = $ifinfo['status'];
-                break;
-            case 'associated':
-                $interfaceItem['status'] = 'up';
-                break;
-            default:
-                $interfaceItem['status'] = '';
-                break;
+    $result = [];
+    $ifsinfo = json_decode((new OPNsense\Core\Backend())->configdpRun('interface list info', [null, true]), true);
```
Comment (fichtner) [inline src/www/widgets/api/plugins/interfaces.inc:34]: Idiom is „ifinfos“ => „ifinfo“ 😉

File: src/www/widgets/api/plugins/interfaces.inc [hunk 3]
```diff
@@ -30,37 +30,31 @@
...
+        if (empty($ifinfo['config']) || (!empty($ifinfo['config']['virtual']))) {
+            /* not configured */
+            continue;
         }
-        $interfaceItem['ipaddr'] = empty($ifinfo['ipaddr']) ? "" : $ifinfo['ipaddr'];
-        $interfaceItem['media'] = empty($ifinfo['media']) ? "" : $ifinfo['media'];
+        $config = $ifinfo['config'];
+        $stats = $ifinfo['statistics'];
+        $interfaceItem = [
+            'inpkts' => $stats['packets received'],
```
Comment (fichtner) [inline src/www/widgets/api/plugins/interfaces.inc:43]: Can we make the widget use flatter by using the internal names?

File: src/www/widgets/include/interface_list.inc [hunk 4]
```diff
@@ -1,4 +1,4 @@
 <?php
 
 $interface_list_title = gettext('Interfaces');
-$interface_list_title_link = 'status_interfaces.php';
+$interface_list_title_link = 'ui/interfaces/overview';
```
Comment (fichtner) [inline src/www/widgets/include/interface_list.inc:4]: Seeing they both point to the same page wouldn’t it make more sense to point this to the assignments page?

File: src/www/widgets/api/plugins/interfaces.inc (hunk 3 above)
Comment (swhite2) [reply src/www/widgets/api/plugins/interfaces.inc:43]: yes, the widget needs a bit more work as it's the last remaining caller of `get_interfaces_info()`.

File: src/opnsense/scripts/interfaces/ifinfo.php (hunk 1 above)
Comment (swhite2) [reply src/opnsense/scripts/interfaces/ifinfo.php:1]: sure :)

File: src/www/widgets/include/interface_list.inc (hunk 4 above)
Comment (swhite2) [reply src/www/widgets/include/interface_list.inc:4]: Good idea :)

**Comment by swhite2** [comment]
note to self: `get_interfaces_info()` and `format_bytes` (`util`) can be cleaned up once `status_overview.php` has been removed.

The requirements to format bytes has appeared at least 3 times now in javascript code, it probably makes sense to move this one level up to the default includes

File: src/www/widgets/widgets/interface_list.widget.php [hunk 5]
```diff
@@ -112,61 +145,27 @@ function interface_widget_update(sender, data)
...
+?>
+  <tr id="interface_widget_item_<?= html_safe($ident) ?>">
+    <td style="width:5%; word-break: break-word;" id="link_status"></td>
+    <td style="width:15%; word-break: break-word;" id="if_name"></td>
+    <td style="width:5%; word-break: break-word;" id="if_status"></td>
+    <td style="width:32%; word-break: break-word;" id="if_media"></td>
     <td style="width:43%; word-break: break-word;">
-      <?= html_safe($ifinfo['ipaddr']) ?>
-      <?= !empty($ifinfo['ipaddr']) ? '<br/>' : '' ?>
-      <?= html_safe(interfaces_has_prefix_only($ifdescr) ? $ifinfo['linklocal'] : $ifinfo['ipaddrv6']) ?>
```
Comment (swhite2) [inline src/www/widgets/widgets/interface_list.widget.php:169]: @fichtner this one is annoying, any particular opinion about it?

Comment (fichtner) [reply src/www/widgets/widgets/interface_list.widget.php:169]: interfaces_primary_address6() should do exactly this to be honest

**Review by fichtner — APPROVED** [review]
Feel free to merge this to progress and make it easier to test / review UI/UX

**Review by AdSchellevis — APPROVED** [review]

**Comment by ohartl** [comment]
This implements #7019 doesn't it?

**Comment by AdSchellevis** [comment]
@ohartl it does, together with https://github.com/opnsense/core/commit/4d495ea6cc3d3c993662867bf3b3a072069e1fc8 to delete the old cruft.

**Pull request merged**